manager/logbroker/broker.go

```diff
@@ -23,25 +24,30 @@ var (
 	errNotRunning     = errors.New("broker is not running")
 )
 
-// LogBroker coordinates log subscriptions to services and tasks. Çlients can
+// LogBroker coordinates log subscriptions to services and tasks. Clients can
 // publish and subscribe to logs channels.
 //
 // Log subscriptions are pushed to the work nodes by creating log subscsription
 // tasks. As such, the LogBroker also acts as an orchestrator of these tasks.
 type LogBroker struct {
-	mu                sync.RWMutex
-	logQueue          *watch.Queue
-	subscriptionQueue *watch.Queue
+	mu                       sync.RWMutex
+	logQueue                 *watch.Queue
+	newSubscriptionsQueue    *watch.Queue
+	changeSubscriptionsQueue *watch.Queue
```

Comment:
The problem though is, by doing that, we'd send down subscriptions multiple times (e.g. subscription now matches node-2 in addition to node-1, matcher.Match(node-1) would return true and therefore we'd send it down again).

I guess this could be solved by having WatchSubscriptions keep a list of subscriptions already sent to the worker and avoid dupes. 

Thoughts?